src/abstractenvironment.jl

```diff
@@ -579,7 +586,8 @@ function test_env(etype::Type{<:AbstractEnvironment}, args...; kwargs...)
 
                     @test getstate(e1) != getstate(e2)
                     @test getaction(e1) == getaction(e2)
-                    @test getobs(e1) != getobs(e2)
+                    @test length(getobs(e1)) == 0 || (getobs(e1) != getobs(e2))
```

Comment:
not really sure what's going on here, two zero length vectors (even of different types) are always equal?